challenge_storing_groceries/src/challenge_storing_groceries/manipulate_machine.py

```diff
@@ -132,13 +169,54 @@ def check_similar_item(userdata=None):
                                                 "no_similar_item": "STORE_ANYWHERE"})
 
             smach.StateMachine.add('STORE_NEAR_ITEM',
-                                   StoreSingleItem(robot, item_designator, place_near_designator),
-                                   transitions={'succeeded': 'ITERATE_ENTITY',
-                                                'failed': 'ITERATE_ENTITY'}
+                                   StoreSingleItem(robot, closest_item_designator, place_near_designator, room=room),
+                                   transitions={'succeeded': 'INSPECT',
+                                                'failed': 'INSPECT'}
                                    )
 
+            @smach.cb_interface(outcomes=["done"])
+            def add_item_to_blacklist(userdata=None):
+                current_blacklist.append(closest_item_designator.resolve().uuid)
+                return "done"
+
+            smach.StateMachine.add("ADD_ITEM_TO_BLACKLIST",
+                                   smach.CBState(add_item_to_blacklist),
+                                   transitions={"done": "INSPECT"})
+
             smach.StateMachine.add('STORE_ANYWHERE',
-                                   StoreSingleItem(robot, item_designator, place_anywhere_designator, arm),
-                                   transitions={'succeeded': 'ITERATE_ENTITY',
-                                                'failed': 'ITERATE_ENTITY'}
+                                   StoreSingleItem(robot,
+                                                   closest_item_designator,
+                                                   place_anywhere_designator,
+                                                   arm,
+                                                   room=room),
+                                   transitions={'succeeded': 'INSPECT',
+                                                'failed': 'ADD_ITEM_TO_BLACKLIST'}
                                    )
+
+
+if __name__ == '__main__':
+    import sys
+    from robot_skills.get_robot import get_robot
+    from robot_smach_states.util.designators import EntityByIdDesignator, ArmDesignator
+
+    if len(sys.argv) < 3:
+        print(f"usage: python {sys.argv[0]} ROBOT ENTITY_ID [Optional: shelf ID]")
+        sys.exit()
+
+    place_area = "on_top_of"
+    if len(sys.argv) > 3:
+        place_area = sys.argv[3]
+
+    rospy.init_node('test_inspect_shelves')
+
+    robot = get_robot(sys.argv[1])
+    robot.reset_all_arms()
+
+    entityDes = EntityByIdDesignator(robot, uuid=sys.argv[2])
+
+    shelfDes = EntityByIdDesignator(robot, uuid="cabinet")
+    armDes = ArmDesignator(robot)
+    placePoseDes = EmptySpotDesignator(robot, shelfDes, armDes, area=place_area, name="empty_spot_designator")
+
+    sm = StoreSingleItem(robot, entityDes, placePoseDes)
+    sm.execute()
```

Comment:
double check names and usage string. Or remove here and fix in https://github.com/tue-robotics/tue_robocup/pull/1270